terminal/runereader_posix.go

```diff
@@ -76,7 +76,7 @@ func (rr *RuneReader) ReadRune() (rune, int, error) {
 		if err != nil {
 			return r, size, err
 		}
-		if r != '[' {
+		if r != '[' && r != 'O' {
```

Comment:
While you're here, you might want to:
- add a link to a terminal reference such as https://vt100.net/docs/vt102-ug/appendixc.html (for VT100).
- comment on each of the switch cases what sequence it is intended to recognize
  e.g. `case 'A': // ESC [ A or ESC O A`
- reject illegal sequences such as 'ESC O 3'. That means (in effect) using different switch cases depending on whether the prefix was ESC [ or ESC O.
- lowercase the error message (and put the word "terminal" in there): "unexpected escape sequence from terminal: %q"

I confirm that this PR fixes the problem I referred to in https://github.com/AlecAivazis/survey/issues/228#issuecomment-915493711. Thanks!